metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -135,19 +161,22 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                     f"Exception while processing table {dataset_path}, skipping it.",
                 )
 
+        # Clean up stale entities at the end
+        yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
```

Comment:
instead of calling add_entity_to_state in a bunch of places, I'd recommend using the automatic wrapper classes

The tableau source is a good example: https://github.com/datahub-project/datahub/blob/44cfd21a653e23f869526a3c1a83e11d51b42a86/metadata-ingestion/src/datahub/ingestion/source/tableau.py#L1615-L1622

You'd need to rename your existing get_workunits() method to get_workunits_internal(), and then add the above code as the get_workunits method. 

Once you do that, you won't need the calls to add_entity_to_state or gen_removed_entity_workunits since it'll be done for you.

Reply:
That definitely simplifies the code quite a bit! Thank you for the recommendation, I added the code changes in the most recent commit.